### PR TITLE
some class definition that related to Address

### PR DESCRIPTION
**Pull request by peggy-gov-bc-ca**
# Description

some class definition that related to Address for Dynamics

## Type of change

Please delete options that are not relevant.
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

tested locally

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: FAMS-761(Create the structure for Address to be sent back to dynamics adapter)

### CODE REVIEW

File: app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/Models/Address.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DynamicsAdapter.Web.PersonSearch.Models
+{
+    public class Address 
+    {
+        public string AddressType { get; set; }
+        public string AddressLine1 { get; set; }
```
Comment (alexjoybc) [inline app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/Models/Address.cs]: should it be StreetLine1 ?

File: app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/Models/Address.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DynamicsAdapter.Web.PersonSearch.Models
+{
+    public class Address 
+    {
+        public string AddressType { get; set; }
```
Comment (alexjoybc) [inline app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/Models/Address.cs]: just Type ?

File: app/DynamicsAdapter/Fams3Adapter.Dynamics/Address/SSG_Address.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        public string AddressLine2 { get; set; }
+
+        [JsonProperty("ssg_locationcanadianprovincecode")]
+        public int Province { get; set; }
+
+        [JsonProperty("ssg_locationcityname")]
+        public string City { get; set; }
+
+        [JsonProperty("ssg_locationcountry")]
+        public int SSG_Country { get; set; }
```
Comment (alexjoybc) [inline app/DynamicsAdapter/Fams3Adapter.Dynamics/Address/SSG_Address.cs]: Remove SSG_

File: app/DynamicsAdapter/Fams3Adapter.Dynamics/Address/SSG_Address.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        public string Name { get; set; }
+
+        [JsonProperty("ssg_locationpostalcode")]
+        public string PostalCode { get; set; }
+
+        [JsonProperty("ssg_locationstatename")]
+        public string NonCanadianState { get; set; }
+
+        [JsonProperty("ssg_SearchRequest")]
+        public virtual SSG_SearchRequest SSG_SearchRequest { get; set; }
```
Comment (alexjoybc) [inline app/DynamicsAdapter/Fams3Adapter.Dynamics/Address/SSG_Address.cs]: remove the SSG_

File: app/DynamicsAdapter/Fams3Adapter.Dynamics/Address/SSG_Address.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
+﻿using Fams3Adapter.Dynamics.SearchRequest;
+using Newtonsoft.Json;
+
+namespace Fams3Adapter.Dynamics.Address
+{
+    public class SSG_Address : DynamicsEntity
+    {
+        [JsonProperty("ssg_addressfulltext")]
+        public string AddressFullText { get; set; }
```
Comment (alexjoybc) [inline app/DynamicsAdapter/Fams3Adapter.Dynamics/Address/SSG_Address.cs]: remove Address

File: app/DynamicsAdapter/Fams3Adapter.Dynamics/Address/SSG_Address.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+        [JsonProperty("ssg_addresscategorytext")]
+        public int AddressCategory { get; set; }
```
Comment (alexjoybc) [inline app/DynamicsAdapter/Fams3Adapter.Dynamics/Address/SSG_Address.cs]: remove Address

File: app/DynamicsAdapter/Fams3Adapter.Dynamics/DynamicsEntity.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Fams3Adapter.Dynamics
+{
+    public class DynamicsEntity
```
Comment (alexjoybc) [inline app/DynamicsAdapter/Fams3Adapter.Dynamics/DynamicsEntity.cs]: this should be abstract class

File: app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/Models/Address.cs (hunk 1 above)
Comment (alexjoybc) [reply app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/Models/Address.cs]: Ignore this one

File: app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/Models/Address.cs (hunk 2 above)
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/Models/Address.cs]: ok

**Pull request merged**